components/tdisplays3/t_display_s3.cpp

```diff
@@ -8,26 +8,132 @@ namespace tdisplays3 {
 
 static const char *const TAG = "TDisplayS3";
 
+#ifdef TDISPLAYS3_USE_ASYNC_IO
+namespace writer_shim {
+  static TaskHandle_t writeFrameTask_ = nullptr;
+  static int16_t x1 = 0;
+  static int16_t x2 = 0;
+  static int16_t y1 = 0;
+  static int16_t y2 = 0;
+  static uint16_t *buffer = nullptr;
+  static void (*callback)(void) = nullptr;
+
+  // A background task used to write the framebuffer to the display
+  static void IRAM_ATTR write_framebuffer_task(void *pv_params) {
+
+    auto tft = (TFT_eSPI *) pv_params;
+
+    while (true) {
```

Comment:
Will the task run the whole time? Is there something that avoids wasting CPU in case the loop finishes "early", or will it just increase the FPS?

`while (true)` loops without any sleep in it finds kinda wrong (busyloop).

Reply:
Makes sense. I haven't implemented tasks (yet) in ESP32, so didn't know how this works. Thanks for the explanation!